VegProcessor/species_hsi/blhwva.py

```diff
@@ -363,55 +363,45 @@ def calculate_si_4(self) -> np.ndarray:
             )
             si_4[~np.isnan(si_4)] = 1
 
-        else:
-            # self.v4a_flood_duration and self.v4b_flow_exchange are NumPy arrays of strings
-            # condition 1
-            mask_1 = (self.v4a_flood_duration == "Temporary") & (
-                self.v4b_flow_exchange == "High"
-            )
-            si_4[mask_1] = 1
-
-            # condition 2
-            mask_2 = (self.v4a_flood_duration == "Seasonal") & (
-                self.v4b_flow_exchange == "High"
-            )
-            si_4[mask_2] = 0.85
-
-            # condition 3
-            mask_3 = (self.v4a_flood_duration == "Semi-Permanent") & (
-                self.v4b_flow_exchange == "High"
-            )
-            si_4[mask_3] = 0.75
-
-            # condition 4
-            mask_4 = (self.v4a_flood_duration == "No Flooding") & (
-                self.v4b_flow_exchange == "High"
-            )
-            si_4[mask_4] = 0.65
-
-            # condition 5
-            mask_5 = (self.v4a_flood_duration == "Temporary") & (
-                self.v4b_flow_exchange == "None"
-            )
-            si_4[mask_5] = 0.5
-
-            # condition 6
-            mask_6 = (self.v4a_flood_duration == "Seasonal") & (
-                self.v4b_flow_exchange == "None"
-            )
-            si_4[mask_6] = 0.4
-
-            # condition 7
-            mask_7 = (self.v4a_flood_duration == "Semi-Permanent") & (
-                self.v4b_flow_exchange == "None"
-            )
-            si_4[mask_7] = 0.25
-
-            # condition 8
-            mask_8 = (self.v4a_flood_duration == "No Flooding") & (
-                self.v4b_flow_exchange == "None"
-            )
-            si_4[mask_8] = 0.1
+        else: 
+            # scoring for 20 flood duration and 
+            # flow exchange combinations of 
+            # conditions
+            si4_score = {
+                ("None", "High"): 0.65,
```

Comment:
We will likely need to map these category string to integers. Using the raw strings will be much less efficient. I'm fine to leave this for now and we will update it when we get flow data.